File: practica2_839756_842236/c++/instruccionPush.cc
// Autores: Andrei Vlasceanu [839756] & Andres Yubero [842236]

#include "instruccionPush.h"

InstruccionPush::InstruccionPush(int _c)
{
    c = _c;
}

void InstruccionPush::ejecutar(stack<int> *s)
{
    s->push(c);
}

void InstruccionPush::listar()
{
    cout << "push " << c << endl;
}

File: practica2_839756_842236/java/InstruccionPush.java
import java.util.Stack;

public class InstruccionPush extends Instruccion {
    private int c;

    InstruccionPush(int _c) {
        c = _c;
    }

    @Override
    public void ejecutar(Stack<Integer> s) {
        s.push(c);
    }

    @Override
    public void listar() {
        System.out.println("push " + c);
    }
}
